xchange-coingi/src/main/java/org/knowm/xchange/coingi/dto/trade/PaginatedResultList.java

```diff
@@ -0,0 +1,44 @@
+package org.knowm.xchange.coingi.dto.trade;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+
+public abstract class PaginatedResultList<T> implements Iterable<T> {
```

Comment:
this and the ResultList look like overengineering to me but if you find them helpful then ok

Reply:
they are neat :)